sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JdbcUtils.scala

```diff
@@ -673,12 +673,12 @@ object JdbcUtils extends Logging {
           stmt.addBatch()
           rowCount += 1
           if (rowCount % batchSize == 0) {
-            stmt.executeBatch()
+            totalUpdatedRows += stmt.executeBatch().sum
             rowCount = 0
           }
         }
         if (rowCount > 0) {
-          stmt.executeBatch()
+          totalUpdatedRows += stmt.executeBatch().sum
```

Comment:
We cannot just sum up `rowCont`?

Reply:
I took the approach to ensure we only count for actual updates, but not sure how Spark has been doing for others. Same for number of bytes written. I was actually asked to update number of bytes written as well, but there's no way to get the actual value from JDBC, so skipped it. 

Please let me know how Spark has been updating these metrics - I'll follow the approach. Thanks!

Reply:
`SparkHadoopWriter` uses a row count as the metric: https://github.com/apache/spark/blob/c128ac564d198effe6bb9754489ea32133dfeb89/core/src/main/scala/org/apache/spark/internal/io/SparkHadoopWriter.scala#L136

Since the returned values of `stmt.executeBatch` seems to be JDBC implementation specific, IMHO its ok to just do the same with `SparkHadoopWriter`

Reply:
OK. Thanks for the guide! What about number of bytes? Reading the length of file is easy, but measuring the size of row for every rows seems nontrivial.